pandas/tests/groupby/aggregate/test_aggregate.py

```diff
@@ -1663,3 +1663,331 @@ def func(x):
     msg = "length must not be 0"
     with pytest.raises(ValueError, match=msg):
         df.groupby("A", observed=False).agg(func)
+
+
+def test_agg_simple_lambda_numpy_to_same_data_type():
+    df = DataFrame(
+        {"A": [1, 3, 100, 3, 100, 100], "B": [False, False, False, False, False, True]}
+    )
+    df["B"] = df["B"].astype("bool")
+    gb = df.groupby("A")
+    result = gb.agg(lambda x: x.max())
+
+    expected = DataFrame({"A": [1, 3, 100], "B": [False, False, True]})
+    expected["B"] = expected["B"].astype("bool")
+    expected.set_index("A", inplace=True)
+    tm.assert_frame_equal(result, expected)
+    assert result["B"].dtype == expected["B"].dtype
```

Comment:
This last assert should not be needed, no? Same for the rest of the tests below.